content/documentation/3-stream-developer-guides/5-troubleshooting/1-debugging-apps.md

```diff
@@ -0,0 +1,182 @@
+Ensure the middleware is available and all configuration is correct.
+Configuration includes credentials, queue/topic names, hosts, etc.
+By default, if no binder dependency is included in the project's classpath, the `TestBinder` will be used.
+
+# Sources
+
+```mermaid
+graph TD;
+    A(Not Producing Data) --> B(Verify External Service Accessibility);
+    B --> C{EnableBinding Annotated?}
+    C --> |Yes| D(Handler Method Defined?)
+    D --> |No| D
+    D --> |Yes| E(Enable SI logging for output channel)
+    C --> |No| C
+```
+
+Sources obtain data from an input and send data to an output channel for downstream processing.
+Ensure the proper class level annotation `@EnableBinding(Source.class)` is present and the handler method is implemented.
+The handler method should use an output channel of `Source.OUTPUT`.
+
+# Processors - Input
+
+```mermaid
+graph TD;
+    A(No Input Data) --> B{EnableProcessor Annotated?};
+    B --> |Yes| C{StreamListener Annotated?}
+    B --> |No| B
+    C --> |Yes| D(Enable SI logging for input channel)
+    C --> |No| C
+```
+
+Processors obtain data, manipulate and return that data for further downstream processing.
+Ensure the proper class level annotation `@EnableBinding(Processor.class)` is present and the handler method is implemented.
+The handler method should use an input channel of `Source.INPUT`.
+
+# Processors - Output
+
+```mermaid
+graph TD;
+    A(No Output Data) --> B{EnableProcessor Annotated?};
+    B --> |Yes| C{SendTo Annotated?}
+    B --> |No| B
+    C --> |Yes| D(Enable SI logging for input channel)
+    C --> |No| C
+```
+
+Processors obtain data, process and return that data for further downstream processing.
+Ensure the proper class level annotation `@EnableBinding(Processor.class)` is present and the handler method is implemented.
+The handler method should use an output channel of `Source.OUTPUT`.
+
+# Sinks
+
+```mermaid
+graph TD;
+    A(Not Receiving Data) --> B{EnableSink Annotated?};
+    B --> |Yes| C{StreamListener Annotated?}
+    B --> |No| B
+    C --> |Yes| E{External Service Accessible?}
+    C --> |No| C
+    E --> |Yes| F(Enable SI logging for input channel)
+    E --> |No| E
+```
+
+Sinks obtain data from an input channel and for example store the that data in an external repository.
+Ensure the proper class level annotation `@EnableBinding(Sink.class)` is present and the handler method is implemented.
+The handler method should use an output channel of `Source.INPUT`.
+
+# Debugging Cloud Foundry Deployments
+
+```mermaid
+graph TD;
+    A(Verify Manifest) --> B(Verify environment variables);
+    B --> C(Services to bind created?)
+    C --> |No| C
+    C --> |Yes| D(Errors in application log files?)
+    D --> |Yes| E(Resolve exceptions)
+    D --> |No| F(Enable SI logging for appropriate channels)
+```
+
+If an application that runs locally, but fails when deployed to Cloud Foundry, first inspect the deployment manifest for correctness.
```

Comment:
maybe add `manifest.yml` in parenthesis since that is a common thing for CF audience to relate to